NeuroID/DataStore.swift

```diff
@@ -63,14 +64,27 @@ public enum DataStore {
     }
 
     static func insertCleanedEvent(event: NIDEvent, storeType: String) {
+    
+        // If queue has more than 2000 events, send a queue full event and return
+        if (DataStore.queuedEvents.count >= max_event_size || DataStore.events.count >= max_event_size) {
+            if (DataStore.events.last?.type != NIDEventName.bufferFull.rawValue) {
+                var fullEvent = NIDEvent.init(type: NIDEventName.bufferFull)
+                if storeType == "queue" {
+                    DataStore.queuedEvents.append(fullEvent)
+                } else {
+                    DataStore.events.append(fullEvent)
+                }
+            }
+            return
+        }
+        
```

Comment:
I thought about this and I think we should move this logic to the clean and store event method instead. because if we are trying to use the least amount of memory possible we should prevent any event work from happening when it issent rather than go and clean the event and then say no we aren't storing it

Reply:
Nice, I like that idea